generators/E906LegacyVtxGen/SQPrimaryVertexGen.h

```diff
@@ -0,0 +1,106 @@
+/*==========================================================================
+  Author: Abinash Pun, Kun Liu
+  Sep, 2019
+  Goal: Import the primary vertex generator of E906 experiment(DPVertexGenerator)
+  from Kun to E1039 experiment in Fun4All framework
+  ============================================================================*/
+
+#ifndef __SQPRIMARYVERTEXGEN_H__
+#define __SQPRIMARYVERTEXGEN_H__
+
+#include <vector>
+#include <memory>
+#include <TString.h>
+#include <TGeoManager.h>
+#include <TF2.h>
+#include <TVector3.h>
+#include <TRandom1.h>
+#include <fun4all/SubsysReco.h>
+#include "SQBeamlineObject.h"
+
+class PHCompositeNode;
+
+class MaterialProfile
+{
+ public:
+  MaterialProfile();
+  int findInteractingPiece(double rndm);
+  void calcProb();
+
+ public:
+  unsigned int nPieces;
+  double probSum;
+  double accumulatedProbs[100];
+  std::vector<SQBeamlineObject> interactables;
+};
+
+class SQPrimaryVertexGen: public SubsysReco
```

Comment:
Do we still want this inheritance?  Will `SQPrimaryVertexGen` be registered to Fun4AllServer in some use cases?

Reply:
I tried to make it a subsysreco class initially but later did not really needed to register to Fun4All. In principle, it should work out without this inheritance.  But somehow, the macro is crashing when I removed it by giving the following error message;
"Fun4AllServer::setRun(): could not get timestamp for run  0, using tics(0) "
I will look into the details and come back to it.

Reply:
That error message always appears at present.  So the crash should be caused by somewhere else.  It's great if you could remove this inheritance, in order to avoid misuse of this class.

Reply:
I will look into it thank you.

Reply:
The fresh build worked. Now it is not a SubsysReco class